src/layman/user/util.py

```diff
@@ -95,3 +98,16 @@ def get_username_suggestions_from_claims(claims):
     if email is not None:
         result.append(email.split('@')[0])
     return result
+
+
+def is_allowed_to_delete_user(username=None):
```

Comment:
Please focus only on self-deletion in this PR. Keep deleting other users for another PR. So remove this method now.